src/main/java/duke/logic/commands/UpdateWeightCommand.java

```diff
@@ -66,20 +66,27 @@ public void stage0(User user, Storage storage) {
         ui.showLine();
         HashMap<LocalDate, Double> allWeight = user.getAllWeight();
         if (!allWeight.containsKey(currentDate)) {
-            try {
-                user.setWeight(Integer.parseInt(weight), currentDate);
-                ui.showWeightUpdate(user, Integer.parseInt(weight), currentDate);
-                ui.showLine();
-            } catch (NumberFormatException e) {
-                ui.showMessage("Please input a proper number for weight");
+            if (Integer.parseInt(weight) > 2) {
```

Comment:
For better code quality, you can use guard clauses to check for invalid lines first.